area_detector_handlers/eiger.py

```diff
@@ -0,0 +1,160 @@
+        'pixel_mask': 'entry/instrument/detector/detectorSpecific/pixel_mask',
+    }
+    specs = {'AD_EIGER2', 'AD_EIGER'}
+
+    def __init__(self, fpath, images_per_file=None, frame_per_point=None):
+        '''
+        Initializer for Eiger handler.
+
+        Parameters
+        ----------
+        fpath : str
+            the partial file path
+
+        images_per_file : int, optional
+            images per file. If not set, must set frame_per_point
+
+        frame_per_point : int, optional. If not set, must set
+            images_per_file
+
+        This one is backwards compatible for both versions of resources
+        saved in databroker. Old resources used 'frame_per_point' as a
+        kwarg. Newer resources call this 'images_per_file'.
+        '''
+        self._file_prefix = fpath
+        if images_per_file is None and frame_per_point is None:
+            raise ValueError(
+                "Either images_per_file or frame_per_point must be not None.")
+
+        if images_per_file is None:
+            # then grab from frame_per_point
+            images_per_file = frame_per_point
+        self._images_per_file = images_per_file
+        self._files = {}
+
+    def __call__(self, seq_id, frame_num=None):
+        '''
+        This returns data contained in the file.
+
+        Parameters
+        ----------
+        seq_id: int
+            The sequence id of the data
+
+        frame_num: int or None
+            If not None, return the frame_num'th image from this
+            3D array. Useful for when an event is one image rather
+            than a stack. (Editor's note: It's not clear what this original
+            docstring was supposed to mean. Is it *ever* not None?)
+
+        Returns
+        -------
+            A dask array
+        '''
+        master_path = Path(f'{self._file_prefix}_{seq_id}_master.h5').absolute()
+        try:
+            file = self._files[master_path]
+        except KeyError:
+            file = h5py.File(master_path, 'r')
+            self._files[master_path] = file
+
+        # TODO This should be captured in documents, not extracted here.
+        # This code is retained just in case, but it does not do anything other
+        # than set self._md, which has no effect unless the user obtain direct
+        # access to the handler instance by reaching into the Filler's cache of
+        # them.
+        md = {k: file[v][()] for k, v in self.EIGER_MD_LAYOUT.items()}
+        # the pixel mask from the eiger contains:
+        # 1  -- gap
+        # 2  -- dead
+        # 4  -- under-responsive
+        # 8  -- over-responsive
+        # 16 -- noisy
+        pixel_mask = md['pixel_mask']
+        # pixel_mask[pixel_mask>0] = 1
+        # pixel_mask[pixel_mask==0] = 2
+        # pixel_mask[pixel_mask==1] = 0
+        # pixel_mask[pixel_mask==2] = 1
+        md['binary_mask'] = (pixel_mask == 0)
+        md['framerate'] = 1. / md['frame_time']
```

Comment:
Should `md['frame_time']` be float-ed in case it's a string in there?
```suggestion
        md['framerate'] = 1. / float(md['frame_time'])
```